### ISSUE TITLE
Primary error message is not shown in error header, instead says `error in 'error' step`

### ISSUE THREAD

**Issue by justinmchase** [jenkinsci/checks-api-plugin#124]
### Feature Request

When I report an error like this in one of my steps:
```groovy
stage ("Submodule Check") {
  steps {
    script {
      error "Submodules are OK for branches but cannot be merged into main"
    }
  }
}
```

This plugin does not report that error cleanly in the check. I expect it to show the message I reported more prominently but instead it shows information about the error step itself.

For example right in the Pull Request instead of showing that error message it simply says `error in 'error' step`:

![Screen Shot 2021-09-04 at 8 10 00 AM](https://user-images.githubusercontent.com/10974/132095706-42ca196d-40c8-4434-a174-d23121b88228.png)

Other github status checkers will show the actual error message in this part of the UI instead.

Additionaly when going to the error details it has the same message prominently in a header as well as some redundant information about an Error in the error step.

(red for bad, green for good)
![Screen Shot 2021-09-04 at 8 11 19 AM](https://user-images.githubusercontent.com/10974/132095868-aca41b76-8290-478d-a280-c9f4b1c89e3c.png)

I feel like in the case of reporting an error via the `error()` step / function, the information about that step should be omitted and the error message itself should be what is reported.

**Comment by KalleOlaviNiemitalo** [comment]
Thank you, @timja.

[WarningStep](https://github.com/jenkinsci/workflow-basic-steps-plugin/blob/92257e3ec20432a31b083c22687f002054dbb5a4/src/main/java/org/jenkinsci/plugins/workflow/steps/UnstableStep.java#L72-L78) adds a WarningAction.

[ErrorStep](https://github.com/jenkinsci/workflow-basic-steps-plugin/blob/92257e3ec20432a31b083c22687f002054dbb5a4/src/main/java/org/jenkinsci/plugins/workflow/steps/ErrorStep.java#L62-L64) just throws an AbortException.  I guess that is then translated to ErrorAction somewhere in workflow-cps-plugin.

**Comment by KalleOlaviNiemitalo** [comment]
Should the formatting just special-case `flowNode.getDisplayFunctionName()` equal to "error" and "unstable", or is there a better way to decide whether the name and arguments of the step are uninteresting?

**Comment by KalleOlaviNiemitalo** [comment]
[isStageNode](https://github.com/jenkinsci/checks-api-plugin/blob/83a4321ed0d4ae69cfa8e2bb34dbfe7b428556dd/src/main/java/io/jenkins/plugins/checks/status/FlowExecutionAnalyzer.java#L189-L196) already hardcodes the "stage" name, so hardcoding "error" and "unstable" as well would be consistent with that.

I imagine changes:

* In `getPotentialTitle`, if `flowNode` is an "error" step, then get `whereBuildFailed` from the first line of [ErrorAction.getDisplayName()](https://github.com/jenkinsci/workflow-api-plugin/blob/405ec3c9594a66c842f30580c9ae41f5e946dc79/src/main/java/org/jenkinsci/plugins/workflow/actions/ErrorAction.java#L104-L107) instead of formatting.
* In `processErrorOrWarningRow`, if `flowNode` is an "error" or "unstable" step, and there is only one argument, and its value matches `errorAction.getDisplayName()` or `warningAction.getMessage()`, then omit the step name and arguments from the output. Thus, if the "unstable" step somehow throws an actual error, then the step name and arguments would still be included.
